tests/test_core/test_new_sequence.py

```diff
@@ -114,8 +114,13 @@ def test_index_a_seq(cast):
 def test_index_a_seq_float_fail(cast):
     seq = new_moltype.DNA.make_seq(seq="TCCAG")
     index = cast(1)
+
+    # avoid codacy "statement seems to have no effect"
+    def idx_seq():
```

Comment:
When codacy raises an issue that we know is invalid, we can identify the error code and add a comment on the line to force codacy to ignore that line.
Instead, change to the following
```python
with pytest.raises(TypeError):
    seq[index]  # pylint: disable=W0104
```